packages/shared/src/store/storage/upload.ts

```diff
@@ -3,7 +3,7 @@ import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
 import { deSig, formatDa, unixToDa } from '@urbit/aura';
```

Comment:
This file is getting a bit convoluted, especially the main `upload` function. I also wonder whether there's anything that can be factored out of the two upload flows, especially once we're doing resizing etc. Any ideas? Or I'm down to discuss. Nonblocking, but would be nice if we can improve.

Reply:
Yeah strong agree here. I should have added more context in the PR description, but I basically started down this path and then backed away because it felt hard to do without the scope creeping quite a bit.

My main takeaway was that if we're going to spend the time to refactor this, we should try to do so in a way that will set us up for any features we'll want for launch or soon after. The big one being avoiding uploading attachments until after you actually send the message. Doing this in a way where we can still update the UI optimistically and (ideally) handle retries got complicated enough that I didn't wanna bite it off here. This becomes more important if we support multiple attachments or videos which I could see being another thing we do sooner rather than later.

I'll log a ticket laying out some of the areas where I think we need improvement to have good baseline UX  and mention how a refactor here would need to be part of that.